### ISSUE TITLE
es-dev-server compatibility mode messing up script requests

### ISSUE THREAD

**Issue by lakshay35** [open-wc/open-wc#625]
when i run the es-dev-server in compatibility mode using the ```es-dev-server --app-index index-es-dev.html --http2 --node-resolve --open --watch --file-extensions .ts --babel``` command, my script requests for external js scripts gets 'http://localhost:8080' appended to them at the front, which messes up the request. 

This works fine without compatibility mode.

**Comment by LarsDenBakker** [comment]
Do you have some example of what you mean by `external js scripts` ?

**Comment by lakshay35** [comment]
It is a request to a local server to respond with a web component js file. Lit-Element to be specific. When it makes the request it appends the localhost:8080 in front of it. In es-dev-server@1.6.6 this was only happening in compatibility mode but was working fine in normal mode. However, in the latest es-dev-server@1.7.1 this behavior is being experienced in both normal mode and compatibility all mode
![image](https://user-images.githubusercontent.com/10873132/61732246-2bc51700-ad4b-11e9-9ae9-8426feca9e63.png)

**Comment by LarsDenBakker** [comment]
Can you show me what your idnex.html looks like?

**Comment by lakshay35** [comment]
<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover" />
    <meta name="Description" content="Put your description here.">

    <style>
        html, body {
            margin: 0;
            padding: 0;
            font-family: sans-serif;
            background-color: #ededed;
        }
    </style>
    <title>components-lib</title>
    <script type="module" src="https://localhost:44397/Components/810AAE53-FC3D-4661-9F3C-01A9896331A1/Polymer3Source/src/service-connect/1.0.0/service-connect.js"></script>

</head>
<body>
  <service-connect singlePage></service-connect>
</body>

</html>

**Comment by lakshay35** [comment]
```<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover" />
    <meta name="Description" content="Put your description here.">

    <style>
        html, body {
            margin: 0;
            padding: 0;
            font-family: sans-serif;
            background-color: #ededed;
        }
    </style>
    <title>components-lib</title>
    <script type="module" src="https://localhost:44397/Components/810AAE53-FC3D-4661-9F3C-01A9896331A1/Polymer3Source/src/service-connect/1.0.0/service-connect.js"></script>

</head>
<body>
  <service-connect singlePage></service-connect>
</body>

</html>
```

**Comment by LarsDenBakker** [comment]
Ah I see, I was testing with `<script src="...">` but you're using `type="module"` which would indeed be picked up as a regular module.

We can check whether it's a request to an external source, but what's the behavior you're expecting here? Should we just leave the script untouched, or do you expect the code to be compiled as well in compatibility mode?

**Comment by lakshay35** [comment]
Hey sorry for the late reply. Got caught up with stuff at work. I am just expecting the resource to be requested as a module and run the browser. The resources are lit-elements and will be registered to the browser session for use as an html component. 

Initially when I tested with es-dev-server in the previous versions. When I imported js files locally using <script src='...' type='module'></script> it worked properly in the normal and compatibility all mode. However, when I replaced it with an external service that responds with files, I am getting an error because the server is appending localhost:8080 to the front. So what I see is that the es-dev-server is already taking care of making the components work through a module import in legacy and modern browsers. Only the url request needs to be untouched (i.e. not adding the origin of the url to the external request uri).
